packages/loader/container-loader/src/container.ts

```diff
@@ -528,7 +537,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
      * Retrieves the audience associated with the document
      */
     public get audience(): IAudience {
-        return this._audience;
+        return this.protocolHandler.audience;
```

Comment:
Just pointing out a possible issue with this. Now that the audience is part of the protocol handler, if a client attempts to get the audience before the container is connected and the protocol handler created, they'll get an exception from https://github.com/microsoft/FluidFramework/pull/11109/files#diff-c94f4bb2aaac11f9fc45cb8bfd7c09d091d54e38e437b727ac1244f1e8d5cb6cR429.

I'm not sure how common that scenario would be, I'm thinking maybe we could make this a breaking change and return `IAudience | undefined` as the situation is not that _exceptional_. 

The contract is valid imho, as one might try to access uninitialized data and we initialize the protocol later in the lifecycle of the container, but moving the Audience is an change that's internal to FF and technically breaks something that used to work..

Reply:
Correct me if I got it wrong, but there should be no scenario where Container is returned to user without protocol fully instantiated.
Same for container runtime - it's created after protocol is created.

Connection can happen before protocol is created, that's true. I think generic mechanism to deal with these problems - pausing signals / ops processing. But 1-off solution (like you do to populate audience) is Ok'ish too.

Reply:
Please see https://portal.microsofticm.com/imp/v3/incidents/details/315979179/home, the "Attempted to access protocolHandler before it was defined" error. It may be a race condition